src/components/Register/index.jsx

```diff
@@ -0,0 +1,256 @@
+import { Alert, Button, Checkbox, Input } from "antd";
+import Modal from "antd/lib/modal/Modal";
+import React, { useState } from "react";
+import { validateEmail } from "../../functions";
+import { useTranslation } from "react-i18next";
+
+const Register = (props) => {
+  const [t] = useTranslation();
+  const [registerInformation, setRegisterInformation] = useState({
+    fullname: "",
+    age: "",
+    email: "",
+    password: "",
+    agree: false,
+  });
+  const [errors, setErrors] = useState({});
+  const [termsOpen, setTermsOpen] = useState(false);
+  const handleChange = (key, value) => {
+    let newValues = Object.assign({}, registerInformation);
+    newValues[key] = value;
+    setRegisterInformation(newValues);
+  };
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    let { fullname, email, age, password, agree } = registerInformation;
+    let errors = {};
+    if (fullname.trim() === "") {
+      errors["fullname"] = t("register.fullName");
+    }
+    if (email.trim() === "") {
+      errors["email"] = t("register.validEmail");
+    }
+    if (age.trim() === "") {
+      errors["age"] = t("register.age");
+    }
+    if (password.trim().length < 6) {
+      errors["password"] = t("register.passwordCharacter");
+    }
+    if (password.trim() === "") {
+      errors["password"] = t("register.enterPassword");
+    }
+    if (!validateEmail(email)) {
+      errors["email"] = t("register.emailFormat");
+    }
+    if (!agree) {
+      errors["agree"] = t("register.terms");
+    }
+    setErrors(errors);
+    if (Object.keys(errors).length > 0) {
+      return;
+    }
+    props.onSubmit(registerInformation);
+  };
+  const handleGoToLogin = () => {
+    props.onGoToLogin();
+  };
+  const handleFacebookAuth = () => {
+    props.onFacebookAuth();
+  };
+  const handleGoogleAuth = () => {
+    props.onGoogleAuth();
+  };
+  const handleOkTerms = () => {
+    handleChange("agree", true);
+    setTermsOpen(false);
+  };
+  const handleCancelTerms = () => {
+    setTermsOpen(false);
+  };
+
+  return (
+    <form noValidate onSubmit={handleSubmit}>
+      <Modal
+        title={t("register.termsTitle")}
+        visible={termsOpen}
+        onOk={handleOkTerms}
+        onCancel={handleCancelTerms}
+        footer={[
+          <Button key="back" onClick={handleCancelTerms}>
+            {t("register.close")}
+          </Button>,
+          <Button key="submit" type="primary" onClick={handleOkTerms}>
+            {t("register.iAgree")}
+          </Button>,
+        ]}
+      >
+        <ul>
```

Comment:
use an array in the translation. see other people's translations

Reply:
updated